openff/bespokefit/executor/services/qcgenerator/worker.py

```diff
@@ -63,87 +74,231 @@ def _select_atom(atoms: List[Atom]) -> int:
-                isomeric=True, explicit_hydrogens=True, mapped=True
-            )
-        },
-        initial_molecule=[
-            molecule.to_qcschema(conformer=i) for i in range(molecule.n_conformers)
-        ],
-        input_specification=QCInputSpecification(
-            model=task.model,
-            driver=DriverEnum.gradient,
-        ),
-        optimization_spec=OptimizationSpecification(
-            procedure=task.optimization_spec.program,
-            keywords={
-                **task.optimization_spec.dict(exclude={"program", "constraints"}),
-                "program": task.program,
             },
-        ),
-    )
+            initial_molecule=[
+                molecule.to_qcschema(conformer=i) for i in range(molecule.n_conformers)
+            ],
+            input_specification=QCInputSpecification(
+                model=task.model,
+                driver=DriverEnum.gradient,
+            ),
+            optimization_spec=OptimizationSpecification(
+                procedure=task.optimization_spec.program,
+                keywords={
+                    **task.optimization_spec.dict(exclude={"program", "constraints"}),
+                    "program": task.program,
+                },
+            ),
+        )
 
-    return_value = qcengine.compute_procedure(
-        input_schema, "torsiondrive", raise_error=True, local_options=_task_config()
-    )
+        return_value = qcengine.compute_procedure(
+            input_schema,
+            "torsiondrive",
+            raise_error=True,
+            local_options=_task_config(),
+        )
 
-    if isinstance(return_value, TorsionDriveResult):
+        if isinstance(return_value, TorsionDriveResult):
 
-        return_value = TorsionDriveResult(
-            **return_value.dict(exclude={"optimization_history", "stdout", "stderr"}),
-            optimization_history={},
+            return_value = TorsionDriveResult(
+                **return_value.dict(
+                    exclude={"optimization_history", "stdout", "stderr"}
+                ),
+                optimization_history={},
+            )
+        # cache the result of the geometry optimisation only if there is a single point spec
+        if sp_specification is not None:
+            task_id = str(uuid.uuid4())
+            redis_connection.hset("qcgenerator:types", task_id, task.type)
+            redis_connection.hset("qcgenerator:task-ids", task_hash, task_id)
+            # mock a result
+            task_meta = {
+                "status": "SUCCESS",
+                "result": return_value.json(),
+                "traceback": None,
+                "children": [],
+                "date_done": datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%S.%f"),
+                "task_id": task_id,
+            }
+            redis_connection.set(f"celery-task-meta-{task_id}", json.dumps(task_meta))
+            _task_logger.info(f"caching torsiondrive result {task_id}")
```

Comment:
I wonder if it would be better to submit this to celery as two chained tasks:

1) Run the torsion drive
2) Run the single point

In this way we don't need to modify the celery metadata directly. Happy to edit this PR directly if you'd like me to take a stab at this later today 🙂

Reply:
Ahhh yes that could work very well so I guess there would still be an overall cache check in the app like we currently have, then checks added to the worker for each of the separate functions?